cmd/metal-api/internal/metal/size.go

```diff
@@ -95,10 +95,10 @@ nextsize:
 	}
 
 	if len(found) == 0 {
-		return nil, matchlog, NotFound("no size found for hardware (%s)", hardware.ReadableSpec())
+		return UnknownSize, matchlog, NotFound("no size found for hardware (%s)", hardware.ReadableSpec())
 	}
 	if len(found) > 1 {
-		return nil, matchlog, fmt.Errorf("%d sizes found for hardware (%s)", len(found), hardware.ReadableSpec())
```

Comment:
I think in this case (which should actually never happen because we prevent overlapping), it should really just fail.

Reply:
It fails, what else do you mean with failing ?